Objects/typeobject.c

```diff
@@ -3383,19 +3384,18 @@ mro_check(PyTypeObject *type, PyObject *mro)
     for (i = 0; i < n; i++) {
         PyObject *obj = PyTuple_GET_ITEM(mro, i);
         if (!PyType_Check(obj)) {
-            PyErr_Format(
-                PyExc_TypeError,
-                "mro() returned a non-class ('%.500s')",
-                Py_TYPE(obj)->tp_name);
+            PyErr_Format(PyExc_TypeError,
+                         "%s.mro() must return class (not %T)",
```

Comment:
This is confusing as `mro()` returns a tuple of classes, not a single class. I think the existing error is better.